### PR TITLE
add slots support for builder and mitosis jsx

### PR DESCRIPTION
**Pull request by steve8708**
that way the AI will work correctly with child nodes that are not direct children

needed for components registered with builder that are like

```<Layout top={...} left={...} />```

currently those don't go to mitosis JSX and see the AI cannot "see" these and things break

### CODE REVIEW

File: packages/core/src/generators/builder.ts [hunk 1]
```diff
@@ -260,7 +266,9 @@ export const blockToBuilder = (
 
   if (thisIsComponent) {
     for (const key in json.bindings) {
-      builderBindings[`component.options.${key}`] = json.bindings[key]!.code;
+      if (!json.slots?.[key]) {
```
Comment (samijaber) [inline packages/core/src/generators/builder.ts:269]: yup good catch. unfortunately need to keep a copy of slots in `json.bindings` as well until all generators have migrated to using `json.slots`

Comment (steve8708) [reply packages/core/src/generators/builder.ts:269]: ya, just no need in the builder json right

File: packages/core/src/__tests__/__snapshots__/mitosis.test.ts.snap [hunk 2]
```diff
@@ -1214,7 +1214,10 @@ import ContentSlotCode from \\"./content-slot-jsx.raw\\";
 export default function SlotCode(props) {
   return (
     <div>
-      <ContentSlotCode slotTesting={<div>Hello</div>} />
+      <ContentSlotCode
+        slotTesting={<div>Hello</div>}
+        slotTesting={<div>Hello</div>}
+      />
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/mitosis.test.ts.snap:1220]: i think you have a bug. you might have forgotten to exclude some of the slots in `json.bindings`

**Review by samijaber — APPROVED** [review]
LGTM.

you forgot the changeset, i added it

**Pull request merged**